File: src/main/java/cz/vhromada/catalog/web/controller/AbstractResultController.java
package cz.vhromada.catalog.web.controller;

import cz.vhromada.validation.result.Result;
import cz.vhromada.validation.result.Status;

/**
 * An abstract class represents controller for processing result.
 *
 * @author Vladimir Hromada
 */
public abstract class AbstractResultController {

    /**
     * Process results.
     *
     * @param results results
     * @throws IllegalArgumentException if results aren't OK
     */
    protected static void processResults(final Result<?>... results) {
        final Result<Void> result = new Result<>();
        for (final Result<?> resultItem : results) {
            result.addEvents(resultItem.getEvents());
        }

        if (Status.OK != result.getStatus()) {
            throw new IllegalArgumentException("Operation result with errors. " + result);
        }
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/controller/AbstractResultController.kt
package cz.vhromada.catalog.web.controller

import cz.vhromada.validation.result.Result
import cz.vhromada.validation.result.Status

/**
 * An abstract class represents controller for processing result.
 *
 * @author Vladimir Hromada
 */
abstract class AbstractResultController {

    /**
     * Process results.
     *
     * @param results results
     * @throws IllegalArgumentException if results aren't OK
     */
    protected fun processResults(vararg results: Result<*>) {
        val result = Result<Void>()
        for (resultItem in results) {
            result.addEvents(resultItem.events())
        }

        require(Status.OK == result.status) { "Operation result with errors. $result" }
    }

}
